### ISSUE TITLE
`play` with `composeStories` throws type error on @next

### ISSUE THREAD

**Issue by kodai3** [storybookjs/testing-react#132]
**Describe the bug**
When I try to write interactive test as described in the [document](https://github.com/storybookjs/testing-react#interactions-with-play-function).

If you use `composeStory`, it works as expected.
If you use `composeStories`, it shows the type error on render.

```
error TS2345: Argument of type '{ canvasElement: HTMLElement; }' is not assignable to parameter of type 'PlayFunctionContext<ReactRenderer, Partial<{ [x: string]: any; }>>'.
  Type '{ canvasElement: HTMLElement; }' is missing the following properties from type 'StoryIdentifier': componentId, title, kind, id, and 3 more.
```

**To Reproduce**
1. Go to ./example
2. Add test with `composeStories`
	```
	const { StandardEmailFilled } = composeStories(stories);

	test('fills input from play function with composeStories', async () => {
	  const { container } = await render(<StandardEmailFilled />);
	
	  await StandardEmailFilled.play({ canvasElement: container });
	
	  const emailInput = screen.getByTestId('email') as HTMLInputElement;
	  expect(emailInput.value).toBe('michael@chromatic.com');
	});
	```
4. See error on your editor or console

**Expected behavior**
Not throwing error. 
Or maybe the usage is changed ?

**Screenshots**
<img width="831" alt="image" src="https://user-images.githubusercontent.com/33568829/211175615-42f33e33-3447-41ff-bd64-d03901496ae0.png">

**Additional context**
I think it relate to this [commit](https://github.com/storybookjs/testing-react/commit/942de96915cdf2329f36e393538e40bf31995a87)

**Comment by kodai3** [comment]
Now with v7rc using `@storybook/react` still get the type error but differently.

<img width="826" alt="スクリーンショット 2023-03-15 11 24 23" src="https://user-images.githubusercontent.com/33568829/225188734-44ad3278-fb60-4d2d-b78a-e5f1048630fe.png">

story side
<img width="284" alt="image" src="https://user-images.githubusercontent.com/33568829/225188949-830af8b2-66ff-42e8-b321-94fd12fa4819.png">

**Comment by kodai3** [comment]
Closing as fix was shipped @storybook/react@7.0.0-rc.4

https://github.com/storybookjs/storybook/pull/21575
